FordLabs/retroquest
# Move cards between columns

## Overview
This addresses #218 by making it possible to drag active thoughts between topic columns.

This makes the app usable for lean coffee. We like to use it for that at work.

This PR does not have an opinion on how the cards sort in the destination column. It does not support moving completed cards. It does not support moving action items.

### Demo
![image](https://user-images.githubusercontent.com/1149737/115622463-7deaf180-a2bd-11eb-93b1-8e853fa5fcef.png)

### Notes
I had to change the ThoughtColumnController.updateThought and deleteThought methods, and they are long and complicated and were not under test. So I put tests around them.

I considered changing the 'edit' API to update the topic as well, and using that instead of creating 'move'. My instinct was to create a specific one and not change an existing. Then, when it came time to call the move API, the column receiving the drop event does not have the whole thought, only its ID and the new topic. That works perfectly for calling a specific service that only moves the thought.

## Testing Instructions
* Run the app (H2 mode is fine). 
* Create a board, and create some cards. 
* Drag the cards between the three columns.
* See the cards move.
* Refresh the page and see them keep their new columns.
